src/App.tsx

```diff
@@ -1,27 +1,67 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import 'bulma/css/bulma.css';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const handleAllGoodsButton = () => {
+    getAll()
+      .then(goodsFromServer => {
+        if (goodsFromServer.length === 0) {
+          throw new Error('Goods are empty');
```

Comment:
each throwing an error will cause app crashing
probably we don't expect such behaviour

can we process error here somehow in other way? without throwing an error on the client